src/Theme/index.js

```diff
@@ -0,0 +1,125 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { StyleSheet } from "react-native";
+import defaultBranding from "../branding_defaults";
+import isObject from "is-obj";
+
+export const ThemeContext = React.createContext({
+  theme: {}
+});
+
+export class Provider extends Component {
+  themeConfigs = {};
+  branding = defaultBranding;
+
+  constructor(props) {
+    super(props);
+    if (props.branding) {
+      this.branding = mergeDeep(this.branding, props.branding);
+    }
+  }
+
+  updateThemeConfig = (name, themeConfig) => {
+    if (this.props.theme && this.props.theme[name])
+      themeConfig = mergeDeep(themeConfig, this.props.theme[name]);
+
+    // Turn the themeConfig object into a string, then replace and strings that start with @ with a value from the branding config
+    // so `@darkColor` would pull the value of `this.branding.darkColor`
+    // We then convert the result back into an object
+    themeConfig = JSON.parse(
```

Comment:
I was thinking if this is the "best" JavaScript way to solve this. I'm open to discussion. What if the API was like this:

```
class A extends Component {
  static themeConfig = ({ linkColor }) => {
    style: {
      base: {
        fontSize: 14,
        color: linkColor,
      }
    }
  };
}
```

What do you think of this?

Reply:
I thought about doing this. The "issue" here is that there is no ability to ensure variables are used correctly. Thus values could be `undefined` and hard to track down.
using a templeted string has the downside of being "less javascripty" and also less flexible... but we can add tests to confirm correct values are used... open to feedback on this though